src/mono/wasm/runtime/hybrid-globalization/change-case.ts

```diff
@@ -39,12 +61,34 @@ export function mono_wasm_change_case(culture: MonoStringRef, src: number, srcLe
         if (!cultureName)
             throw new Error("Cannot change case, the culture name is null.");
         const input = get_utf16_string(src, srcLength);
-        let result = toUpper ? input.toLocaleUpperCase(cultureName) : input.toLocaleLowerCase(cultureName);
-        if (result.length > destLength)
-            result = input;
+        const result = toUpper ? input.toLocaleUpperCase(cultureName) : input.toLocaleLowerCase(cultureName);
 
-        for (let i = 0; i < destLength; i++)
-            setU16(dst + i*2, result.charCodeAt(i));
+        if (result.length <= destLength)
```

Comment:
would length report 2 for multi-surrogate characters like U+10437 (𐐷) ?

Reply:
All good 
![image](https://github.com/dotnet/runtime/assets/271576/eb5c1fd1-476d-45d6-86c6-10556f3b6ab4)